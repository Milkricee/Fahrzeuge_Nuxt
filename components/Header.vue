<template>
  <header class="bg-primary-50 p-4 shadow-custom">
    <nav class="flex items-center space-x-4">
      <img src="/assets/images/AI_logo.png" alt="Logo" class="h-8 w-auto" />

      <ul class="flex space-x-4">
        <li>
          <NuxtLink to="/kfz/fahrzeuge" 
            class="px-3 py-2 bg-primary-50 rounded hover:bg-primary-200 text-primary-800 hover:text-black transition-colors duration-300 font-bold">
            Fuhrpark
          </NuxtLink>
        </li>
        <li>
          <NuxtLink to="/versicherung" @click.prevent="goToVersicherung" 
            class="px-3 py-2 bg-primary-50 rounded hover:bg-primary-200 text-primary-800 hover:text-black transition-colors duration-300 font-bold">
            Versicherung
          </NuxtLink>
        </li>
      </ul>
    </nav>
  </header>
</template>

<script setup>
import { navigateTo } from '#app';
import { resetFahrzeugList } from '~/composables/fahrzeugHelper'; // Importiere aus fahrzeugHelper.js

async function goToVersicherung() {
  await resetFahrzeugList(); // Warten, bis resetFahrzeugList abgeschlossen ist
  navigateTo('/versicherung');
}
</script>