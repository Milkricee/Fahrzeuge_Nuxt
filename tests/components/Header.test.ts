import { describe, it, expect, vi } from 'vitest';  // Importiere die notwendigen Testmethoden von Vitest
import { mount } from '@vue/test-utils';
import Header from '@/components/Header.vue'; // Stelle sicher, dass der Pfad korrekt ist

// Mocken der Nuxt-Funktionen
vi.mock('#app', () => ({
  navigateTo: vi.fn()  // Erstelle einen Mock für `navigateTo`
}));

describe('Header.vue', () => {
  it('renders the logo with correct alt text', () => {
    const wrapper = mount(Header);

    // Stelle sicher, dass das Logo korrekt angezeigt wird
    const logo = wrapper.find('img');
    expect(logo.exists()).toBe(true);  // Überprüfe, dass das Logo-Element vorhanden ist
    expect(logo.attributes('alt')).toBe('Logo');  // Überprüfe, dass der Alt-Text korrekt ist
  });

  it('matches the header structure with a snapshot', () => {
    const wrapper = mount(Header);

    // Vergleiche die gerenderte Struktur des Headers mit einem Snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('calls navigateTo when clicking on the Versicherung link', async () => {
    const wrapper = mount(Header);

    // Finde den "Versicherung"-Link und klicke darauf
    const versicherungLink = wrapper.find('a[href="/versicherung"]');
    await versicherungLink.trigger('click.prevent');

    // Überprüfe, ob `navigateTo` aufgerufen wurde
    expect(vi.fn().mock.calls.length).toBe(1);  // Überprüfe, dass navigateTo einmal aufgerufen wurde
    expect(vi.fn().mock.calls[0][0]).toBe('/versicherung');  // Überprüfe, dass navigateTo mit dem richtigen Parameter aufgerufen wurde
  });
});
