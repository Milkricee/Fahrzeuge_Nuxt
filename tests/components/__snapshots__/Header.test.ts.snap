// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header.vue > matches the header structure with a snapshot 1`] = `
"<header class="bg-primary-50 p-4 shadow-custom">
  <nav class="flex items-center space-x-4">
    <!-- Logo --><img src="/assets/images/AI_logo.png" alt="Logo" class="h-8 w-auto">
    <ul class="flex space-x-4">
      <li>
        <nuxtlink to="/kfz/fahrzeuge" class="px-3 py-2 bg-primary-50 rounded hover:bg-primary-200 hover:text-black transition-colors duration-300"> Fuhrpark </nuxtlink>
      </li>
      <li>
        <nuxtlink to="/versicherung" class="px-3 py-2 bg-primary-50 rounded hover:bg-primary-200 hover:text-black transition-colors duration-300"> Versicherung </nuxtlink>
      </li>
    </ul>
  </nav>
</header>"
`;
